### ISSUE TITLE
Bugs to support regex crate "0.2"

### ISSUE THREAD

**Issue by cjppku** [Geal/nom#441]
Currently, nom looks like only support regex "^0.1", when I set the version of regex to "0.2", it will report error.

## to find a string:
```rust
    named!(rm(&str) -> &str, re_find!(r"^\d{4}-\d{2}-\d{2}"));
```

```
error[E0308]: mismatched types
  --> src/main.rs:10:5
10 |     named!(rm(&str) -> &str, re_find!(r"^\d{4}-\d{2}-\d{2}"));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `regex::Match`, found tuple
   |
   = note: expected type `regex::Match<'_>`
   = note:    found type `(_, _)`
   = note: this error originates in a macro outside of the current crate
```

## to capture a regex,
```rust
       named!(rm(&str) -> &str, re_capture!(r"^(\d{4})-(\d{2})-(\d{2})"));
```

error message:
```
error: no method named `iter_pos` found for type `regex::Captures<'_>` in the current scope
  --> src/main.rs:10:5
   |
10 |     named!(rm(&str) -> &str, re_capture!(r"^(\d{4})-(\d{2})-(\d{2})"));
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate
```

Please to fixed it, thanks.

**Comment by Geal** [comment]
which version of nom are you using? nom 2.1 uses regex 2.0

**Comment by Geal** [comment]
ok, I see the bug, thanks for the report, this will get in the next nom version

**Comment by Geal** [comment]
fixed by #444
